pimd/pim_igmp.c

```diff
@@ -753,6 +753,36 @@ static void igmp_group_free(struct igmp_group *group)
 	XFREE(MTYPE_PIM_IGMP_GROUP, group);
 }
 
+static void igmp_group_count_incr(struct igmp_sock *igmp)
+{
+	struct pim_interface *pim_ifp = igmp->interface->info;
+
+	if (pim_ifp) {
+		++pim_ifp->pim->igmp_group_count;
+		if (pim_ifp->pim->igmp_group_count
+		    == pim_ifp->pim->igmp_watermark_limit) {
+			zlog_warn(
+				"SCALE ALERT: igmp group count reached watermak limit: %u(vrf: %s)",
+				pim_ifp->pim->igmp_group_count,
+				pim_ifp->pim->vrf->name);
+		}
+	}
+}
+
+static void igmp_group_count_decr(struct igmp_sock *igmp)
+{
+	struct pim_interface *pim_ifp = igmp->interface->info;
+
+	if (pim_ifp) {
+		if (!pim_ifp->pim->igmp_group_count) {
+			zlog_warn(
+				"WARNING: cannot decrement igmp group count below 0");
```

Comment:
wouldn't have to break this line if you had
```
if (!pim_ifp)
   return;
```
at the beginning instead of the whole function body in an `if` block

Reply:
Done